url8db16c7d62_Arsaell_EdT_tgz-pJ8-DATA_MainJ8/src/DATA/Filler.java

```diff
@@ -0,0 +1,980 @@
+				
+				boucle 2 : plus traditionnelle, parcourt pour chaque matière la liste d'enseignants depuis le début.
+			
+		*/
+		boolean failed = false;
+		
+		int i = 0, j = 0;
+		
+		for (Field f : order)	{
+		
+			for (Group g : this.groups)	{
+			
+				if (g.getClasses().containsKey(f))	{
+				
+					j = i;
+					for (i = i ; i < this.teachers.size() ; i++)	{
+					
+						Teacher teach = this.teachers.get(i);
+						
+						if (teach.canTeach(f, g))	{
+						
+							g.setTeacher(teach, f);
+							//ps.println("Filler.attributeTeachers(Field[] order) : " + g + " " + f + " " + teach + " --> " + res);
+							break;
+						}
+						
+						/*
+						else if (teach == this.teachers.get(this.teachers.size() - 1))
+							ps.println("Filler.attributeTeachers(Field[] order) says : Couldn't attribute a teacher for field " +  f + " to group " + g);
+						//*/
+						
+						if(i == this.teachers.size() - 1)
+							i = -1;
+						
+						if (i == j - 1)	{
+							failed = true;
+							//ps.println("Filler.attributeTeachers(Field[] order) says : Couldn't attribute a teacher for field " +  f + " to group " + g);
+							break;
+						}
+					}
+				}
+			}
+		}
+		
+		if (failed)	{
+		
+			failed = false;
+			
+			for (Field f : order)	{
+				
+				for (Group g : this.groups)	{
+				
+					if (g.getClasses().containsKey(f))	{
+					
+						for (Teacher teach : this.teachers)	{
+						
+							if (teach.canTeach(f, g))	{
+							
+								boolean res = g.setTeacher(teach, f);
+								//ps.println("Filler.attributeTeachers(Field[] order) : " + g + " " + f + " " + teach + " --> " + res);
+								break;
+							}
+							
+							
+							else if (teach == this.teachers.get(this.teachers.size() - 1))	{
+								failed = true;
+								ps.println("Filler.attributeTeachers(Field[] order) says : Couldn't attribute a teacher for field " +  f + " to group " + g);
+							}
+						}
+					}
+				}
+			}
+		}
+		
+		return (!failed);
+	}
+	
+	/**
+	 * Cette méthode est appelée peu après l'instanciation du Filler pour déterminer
+	 * dans quel ordre il est préférable de traiter les éléments des listes.
+	 * Il détermine pour chaque classe implémentant l'interface Constrainable 
+	 * une contrainte sous forme d'un double entre 0 et 1 :
+	 * 0 signifiant que l'élément est complètement libre
+	 * 1 signifiant que la marge de manoeuvre lors de l'attribution/placement de l'élément est nulle.
+	 * Elle renvoie une HashMap ordonnée de ces contraintes via la méthode orderByValues()
+	 * @param attributeTeachers détermine si la méthode devrait ou non profiter du calcul des contraintes
+	 * pour tenter d'attribuer les Teachers aux Groups et ainsi ensuite recalculer plus précisément
+	 * les contraintes des Teachers
+	 * @return une liste ordonnée de Constrainables associés de leur contrainte.
+	 */
+	public HashMap<Constrainable, Double> computeConstraints(boolean attributeTeachers)	{
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 43. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 266-280, lines 284-290, lines 292-305 into separate methods.